src/app-layer-smtp.c

```diff
@@ -1334,13 +1388,28 @@ static inline void ResetLine(SMTPLine *line)
  *          1 for handing control over to GetLine
  *         -1 for errors and inconsistent states
  * */
-static int SMTPPreProcessCommands(
-        SMTPState *state, Flow *f, AppLayerParserState *pstate, SMTPInput *input, SMTPLine *line)
+static int SMTPPreProcessCommands(SMTPState *state, Flow *f, AppLayerParserState *pstate,
+        StreamSlice *slice, SMTPInput *input, SMTPLine *line)
 {
     DEBUG_VALIDATE_BUG_ON((state->parser_state & SMTP_PARSER_STATE_COMMAND_DATA_MODE) == 0);
     DEBUG_VALIDATE_BUG_ON(line->len != 0);
     DEBUG_VALIDATE_BUG_ON(line->delim_len != 0);
 
+    Frame *data_frame = NULL;
+    if (state->curr_tx) {
+        data_frame = state->curr_tx->data_frame_opened
+                             ? AppLayerFrameGetLastOpenByType(f, 0, SMTP_FRAME_DATA)
+                             : NULL;
+        if (data_frame == NULL && !state->curr_tx->data_frame_opened) {
+            data_frame = AppLayerFrameNewByPointer(
```

Comment:
Is it not too soon to open this frame ?

Reply:
I don't think so, why do you think it may be?

Reply:
cf usage of `is_data` ?